jdbc-v2/src/main/java/com/clickhouse/jdbc/StatementImpl.java

```diff
@@ -289,6 +301,7 @@ private boolean execute(String sql, QuerySettings settings) throws SQLException
             executeQuery(sql, settings); // keep open to allow getResultSet()
             return true;
         } else if(type == StatementType.SET) {
+            executeUpdate(sql, settings);
```

Comment:
@chernser 
I don't think that we need to execute this update; the role URL setting is enough, as the HTTP interface is stateless.